server/grpc/rpc.go

```diff
@@ -69,6 +67,20 @@ func (s *RPC) Next(c context.Context, agentFilter rpc.Filter) (*rpc.Workflow, er
 		return nil, nil
 	}
 
+	// enforce server set agent filters ...
+	agentServerFilters, err := agent.GetServerFilters()
```

Comment:
We now have two ways for setting filters. IMO we should decided for one: UI or agent flags.
I would currently favor to keep agent flags as those can be easily set by deployment tools like helm, ansible. The ability to switch labels on the fly (without restarting the agent) is interesting, but honestly not really necessary as I can't image someone updating those regularly.

Reply:
moved to https://github.com/woodpecker-ci/woodpecker/pull/4131 to not hold this pull back